src/app/sign-up-metamask/sign-up-metamask.component.ts

```diff
@@ -0,0 +1,255 @@
+    } catch (e) {
+      this.errorMessage = `Can\'t connect to the Metamask API. Error: ${e}. Please try again.`;
+      this.metamaskState = METAMASK.ERROR;
+      return;
+    }
+    // fetch a spending limit hex string based off of the permissions you're allowing
+    const getAccessBytesResponse = await this.backendApi
+      .GetAccessBytes(
+        derivedPublicKeyBase58Check,
+        expirationBlock,
+        getSpendingLimitsForMetamask() as TransactionSpendingLimitResponse
+      )
+      .toPromise()
+      .catch((e) => {
+        this.errorMessage =
+          'Problem getting transaction spending limit. Please try again.';
+        this.metamaskState = METAMASK.ERROR;
+        return;
+      });
+    if (!getAccessBytesResponse) {
+      return;
+    }
+
+    // we can now generate the message and sign it.
+    let message: number[];
+    let signature: string;
+    try {
+      const resp =
+        await this.metamaskService.signMessageWithMetamaskAndGetEthAddress(
+          getAccessBytesResponse.AccessBytesHex
+        );
+      message = resp.message;
+      signature = resp.signature;
+    } catch (e) {
+      this.errorMessage =
+        'Something went wrong while producing Metamask signature. Please try again.';
+      this.metamaskState = METAMASK.ERROR;
+      return;
+    }
+
+    // once we have the signature we can fetch the public key from it
+    const metamaskKeyPair = this.getMetaMaskMasterPublicKeyFromSignature(
+      signature,
+      message
+    );
+
+    const metamaskPublicKey = Buffer.from(
+      metamaskKeyPair.getPublic().encode('array', true)
+    );
+
+    const metamaskPublicKeyHex = metamaskPublicKey.toString('hex');
+    const metamaskBtcAddress = this.cryptoService.publicKeyToBtcAddress(
+      metamaskPublicKey,
+      Network.mainnet
+    );
+
+    const metamaskEthAddress =
+      this.cryptoService.publicKeyToEthAddress(metamaskKeyPair);
+    const metamaskPublicKeyBase58Check =
+      this.cryptoService.publicKeyToDeSoPublicKey(metamaskKeyPair, network);
+    try {
+      await this.backendApi
+        .SendStarterDeSoForMetamaskAccount({
+          Signer: metamaskKeyPair.getPublic().encode('array', true),
+          AmountNanos: 1000,
+          Message: message,
+          Signature: [
+            ...Buffer.from(signature.slice(2, signature.length), 'hex'),
+          ],
+        })
+        .toPromise();
+    } catch (e) {
+      const err = (e as any)?.error?.error;
+      this.errorMessage =
+        err ||
+        'Unable to send starter Deso, this is not an issue if you already have a Deso balance';
+      this.metamaskState = METAMASK.ERROR;
+      return;
+    }
+    // Slice the '0x' prefix from the signature.
+    const accessSignature = signature.slice(2);
+    // we now have all the arguments to generate an authorize derived key transaction
+    let authorizeDerivedKeyResponse: any;
+    try {
+      authorizeDerivedKeyResponse = await this.backendApi
```

Comment:
I think there may be a weird state we can end up in where the sign-up bonus is sent but the authorize derived key transaction gets ejected from the mempool. We need someway for users to be able to still authorize a derived key even if they've received a sign-up vonus.